### PR TITLE
Qg2

### PR DESCRIPTION
**Pull request by OliverSherouse**

### CODE REVIEW

File: docs/corpus.markdown [hunk 1]
```diff
@@ -1,45 +1,166 @@
...
+and department authored each individual CFR part.
 
--   A `makefile` to manage the workflow of the corpus
--   A module named `driver.py` that implements the corpus driver interface (see below)
--   A subdirectory named `scripts` containing the scripts needed to obtain, organize, and prepare the documents that make up the corpus, as well as to generate any metadata for the corpus.
--   A subdirectory named `data` that holds any intermediate data generated during preparation of the corpus.
+## Starting a New Corpus
 
-One file that each corpus should generate is a csv file in the `data` directory named `metadata.csv`. This file should contain any additional information about individual documents that may be relevant. For example, the `metadata.csv` generated by the CFR corpus includes which agency and department authored each individual CFR part, as well as the restriction and word count for each part.
+The fastest way to start a new corpus is to use the command
```
Comment (Stephen-Strosko) [inline docs/corpus.markdown:24]: Do we want to add something here about installing the library first?

File: docs/corpus.markdown [hunk 2]
```diff
@@ -1,45 +1,166 @@
...
+Each corpus should contain a python module named `driver.py`. This driver
+serves two important functions. First it specifies how the corpus should be
+indexed. An index is one or more values that, taken together, uniquely identify
+each document in the corpus. An index can be as simple as an id number, or it
+can be more descriptive. For the CFR corpus, each document, representing a
+single subdivision called a part, is represented by three pieces of metadata:
+the year of the CFR edition in which the part was printed, the to which the
+part belongs, and the part number.
+
+The `driver.py` file should define a variable called `driver` which holds the
```
Comment (Stephen-Strosko) [inline docs/corpus.markdown]: This paragraph on the driver is slightly confusing to me. What is the driver object in this case?

Comment (Stephen-Strosko) [reply docs/corpus.markdown]: I know we give examples below, but maybe just a sentence on what a driver object does generally.

File: docs/corpus.markdown [hunk 3]
```diff
@@ -1,45 +1,166 @@
...
+BASE_DIR = Path(__file__).resolve().parent
 
-The easiest way to write a new corpus is to fork the most similar official corpus (see below) and modify it to represent the relevant body of text. There are three principle problems to solve in the creation of a new corpus:
+driver = qg.corpus.IndexDriver(index=BASE_DIR.joinpath('data', 'index.csv'))
+```
 
-1.  How can the text be obtained and, if necessary, translated to plain text?
-2.  What is the logical unit of analysis for the corpus?
-3.  How can the text be organized to most usefully reflect the unit of analysis.
+### Custom Drivers
```
Comment (Stephen-Strosko) [inline docs/corpus.markdown:147]: I was pretty lost with this paragraph. Maybe we need an example, or just cut it if it is something that we think people will never use.

File: docs/ml.markdown [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+### Initializing the Estimator
+
+To start a new quantgov estimator, use the following command
+`quantgov start estimator myname` where `myname` is the name you want to
+give to the estimator. This will copy the [estimator
+skeleton](https://github.com/quantgov/estimator) to the folder with that
+name.
+
+### Customize the Vectorizer and Vectorize Trainers
+
```
Comment (Stephen-Strosko) [inline docs/ml.markdown:26]: Maybe add one or two sentences that explain what a vectorizer is.

File: docs/nlp.markdown [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+The QuantGov library includes a number of built-in utilities to perform some
+natural language analysis on a corpus. These can be run using the
+`quantgov nlp` set of commands. All nlp analyses are run at the document level,
+and output a csv. By default, results are printed to standard output, but an
+output file may be specified with the `-o` flag.
+
+For example, the command
+
+``` {.bash}
```
Comment (Stephen-Strosko) [inline docs/nlp.markdown:11]: {.bash} ?

File: docs/nlp.markdown [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+be merged with any data or statistical analysis software to create a combined
+dataset.
+
+Note that not all analyses may be aggregated above the document level; users
+should familiarize themselves with the metrics produced and use them
+appropriately.
+
+## Available Analyses
+
+-   **count\_words**: Count the number of words in a document. The regular
```
Comment (Stephen-Strosko) [inline docs/nlp.markdown]: I am not sure what you mean by the "regular expression for a word". Maybe clarify what that regular expression looks to match.

File: docs/tutorial/first_dataset.markdown [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+## Running NLP Analyses
+
+As discussed [earlier](./natural_language_analysis.markdown),
+
+Let's start by running two built-in QuantGov analyses: the standard word
+count, and the conditional count, which gives one measure of the
+complexity of the document. Run these two commands:
+
+``` {.bash}
```
Comment (Stephen-Strosko) [inline docs/tutorial/first_dataset.markdown:15]: I see {.bash} again, this must be something I am not picking up on.

File: docs/corpus.markdown (hunk 1 above)
Comment (OliverSherouse) [reply docs/corpus.markdown:24]: Don't need to do this on this page, since we do it on the front page.

File: docs/corpus.markdown (hunk 2 above)
Comment (OliverSherouse) [reply docs/corpus.markdown]: Updated to clarify.

File: docs/ml.markdown (hunk 4 above)
Comment (OliverSherouse) [reply docs/ml.markdown:26]: Done.

File: docs/nlp.markdown (hunk 5 above)
Comment (OliverSherouse) [reply docs/nlp.markdown:11]: Properly formats the code block.

File: docs/nlp.markdown (hunk 6 above)
Comment (OliverSherouse) [reply docs/nlp.markdown]: Clarified.

File: docs/tutorial/first_dataset.markdown (hunk 7 above)
Comment (OliverSherouse) [reply docs/tutorial/first_dataset.markdown:15]: See above.

**Pull request merged**